Source/Core/DolphinQt2/Config/Graphics/EnhancementsWidget.cpp

```diff
@@ -0,0 +1,260 @@
+      new GraphicsChoice({tr("Off"), tr("Side-by-Side"), tr("Top-and-Bottom"), tr("Anaglyph")},
+                         Config::GFX_STEREO_MODE);
+  m_3d_depth = new GraphicsSlider(0, 100, Config::GFX_STEREO_DEPTH);
+  m_3d_convergence = new GraphicsSlider(0, 200, Config::GFX_STEREO_CONVERGENCE, 100);
+  m_3d_swap_eyes = new GraphicsBool(tr("Swap Eyes"), Config::GFX_STEREO_SWAP_EYES);
+
+  stereoscopy_layout->addWidget(new QLabel(tr("Stereoscopic 3D Mode")), 1, 1);
+  stereoscopy_layout->addWidget(m_3d_mode, 1, 2);
+  stereoscopy_layout->addWidget(new QLabel(tr("Depth")), 2, 1);
+  stereoscopy_layout->addWidget(m_3d_depth, 2, 2);
+  stereoscopy_layout->addWidget(new QLabel(tr("Convergence")), 3, 1);
+  stereoscopy_layout->addWidget(m_3d_convergence, 3, 2);
+  stereoscopy_layout->addWidget(m_3d_swap_eyes, 4, 1);
+
+  AddWidget(enhancements_box);
+  AddWidget(stereoscopy_box);
+
+  AddDescriptionBox();
+}
+
+void EnhancementsWidget::ConnectWidgets()
+{
+  connect(m_aa_combo, static_cast<void (QComboBox::*)(int)>(&QComboBox::currentIndexChanged),
+          [this](int) { SaveSettings(); });
+  connect(m_pp_effect, static_cast<void (QComboBox::*)(int)>(&QComboBox::currentIndexChanged),
+          [this](int) { SaveSettings(); });
+  connect(m_3d_mode, static_cast<void (QComboBox::*)(int)>(&QComboBox::currentIndexChanged),
+          [this](int) { SaveSettings(); });
+}
+
+void EnhancementsWidget::LoadSettings()
+{
+  // Anti-Aliasing
+  m_aa_combo->clear();
+  for (const auto& option : VideoUtils::GetAvailableAntialiasingModes(m_msaa_modes))
+    m_aa_combo->addItem(option == "None" ? tr("None") : QString::fromStdString(option));
+
+  auto selection = g_Config.iMultisamples;
+
+  if (g_Config.backend_info.bSupportsSSAA && g_Config.bSSAA)
+    selection += m_msaa_modes;
+
+  m_aa_combo->setCurrentIndex(selection);
+
+  // Post Processing Shader
+  std::vector<std::string> shaders =
+      g_Config.iStereoMode == STEREO_ANAGLYPH ?
+          PostProcessingShaderImplementation::GetAnaglyphShaderList(
+              g_Config.backend_info.api_type) :
+          PostProcessingShaderImplementation::GetShaderList(g_Config.backend_info.api_type);
+
+  m_pp_effect->clear();
+  m_pp_effect->addItem(tr("(off)"));
+
+  for (const auto& shader : shaders)
+  {
+    m_pp_effect->addItem(QString::fromStdString(shader));
+    if (g_Config.sPostProcessingShader == shader)
+      m_pp_effect->setCurrentIndex(m_pp_effect->count() - 1);
+  }
+
+  PostProcessingShaderConfiguration pp_shader;
+  if (g_Config.sPostProcessingShader != "(off)")
+  {
+    pp_shader.LoadShader(g_Config.sPostProcessingShader);
+    m_configure_pp_effect->setEnabled(pp_shader.HasOptions());
+  }
+}
+
+void EnhancementsWidget::SaveSettings()
+{
+  bool is_ssaa = m_aa_combo->currentText().startsWith(QStringLiteral("SSAA"));
+  g_Config.iMultisamples = m_aa_combo->currentIndex() - (is_ssaa ? m_msaa_modes : 0);
+
+  g_Config.sPostProcessingShader = m_pp_effect->currentText().toStdString();
```

Comment:
Config::SetBaseOrCurrent(Config::GFX_ENHANCE_POST_SHADER, /* ... */);

There do not be appear to be any other instances where you've directly written to `g_Config`.